File: 2306-create-binary-tree-from-descriptions/2306-create-binary-tree-from-descriptions.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode createBinaryTree(int[][] arr) {
        HashMap<Integer, TreeNode> map = new HashMap<>();
        HashSet<Integer> st = new HashSet<>();

        for(int[] it : arr) {
            if(!map.containsKey(it[0])) map.put(it[0], new TreeNode(it[0]));
            if(!map.containsKey(it[1])) map.put(it[1], new TreeNode(it[1]));

            if(it[2] == 1) map.get(it[0]).left = map.get(it[1]);
            else map.get(it[0]).right = map.get(it[1]);
            st.add(it[1]);
        }

        for(int[] it : arr) {
            if(!st.contains(it[0])) return map.get(it[0]);
        }

        return null;
    }
}

File: 2306-create-binary-tree-from-descriptions/2306-create-binary-tree-from-descriptions.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {number[][]} descriptions
 * @return {TreeNode}
 */
var createBinaryTree = function(desc) {
    const mp = new Map();
    const st = new Set();

    for(let it of desc) {
        if(!mp.has(it[0])) mp.set(it[0], new TreeNode(it[0]));
        if(!mp.has(it[1])) mp.set(it[1], new TreeNode(it[1]));
        
        if(it[2] == 1) mp.get(it[0]).left = mp.get(it[1]);
        else mp.get(it[0]).right = mp.get(it[1]);

        st.add(it[1]);
    }

    for(let it of desc) {
        if(!st.has(it[0])) return mp.get(it[0]);
    }

    return null;
};